contribute/process/images/abiprocess.pu

```diff
@@ -0,0 +1,25 @@
+[*] --> NewABIChange: Firmware feature that requires ABI change
+
+NewABIChange --> RFC: Send RFC Pull Request to \nget driver and fw stakeholder feedback
+
+note right of RFC : RFC stage is intended\nto avoid wasted effort via \nearly engagement with the ABI users \n(driver developers)
+
+RFC --> FWImplementation: Ok proceed to implement full change
+
+RFC --> DriverImplementation: Start working on driver changes
+
+NewABIChange --> FWImplementation: Fast path implementation\nfor minor changes
+
+note left of FWImplementation : Fast path only when no impact to driver
+
+FWImplementation --> ABIClassification: Submit PR for classification
+
+ABIClassification -> DriverImplementation: Start working on driver changes
```

Comment:
Why are there two paths marked "Start working on driver changes", leading to "DriverImplementation?" Can you start the same work twice? :-) It also indicates a route from fast path to driver implementation, which shouldn't be there, since the fast path means no driver changes? Also in one note the fast path is described as "no impact to driver" but in another it's "minor changes" which are two different criteria?

Reply:
@lyakh Good question. I This was on purpose. I wanted to highlight that if you go via the "normal path", we can sooner figure out whether a driver change is needed to merge the ABI change. Currenly we have many cases where this realization only happens at the ABI board meeting, which is very late. If you take the "fast path", you have the risk that you will be sent back to the drawing board. Makes sense?

Reply:
In what context is a change to the driver no required? If the ABI is being updated won't it need it in both places? Or is this referring to a backwards compatible change. (e.g. adding to an enum)

Reply:
@cujomalainey wrote:
> In what context is a change to the driver no required? If the ABI is being updated won't it 
> need it in both places? Or is this referring to a backwards compatible change. (e.g. adding 

Indeed these would be MINOR changes, which are backwards compatible. If we bump MAJOR version, then driver change is mandatory.

What has been muddying the picture are MINOR ABI changes, but where part of the FW change takes new features into use in the SOF topologies used in CI. This was now taken out from the ABI process description as it's not really ABI management anymore, but a question of how the CI(s) is/are managed. In many cases driver change is still optional, but e.g. if you now add a new component type (pre-UUID system), kernel must be modified to understand the new component. With the UUID based component creation in place, we will have less and less cases like this.